timesync/manual_clock.go

```diff
@@ -0,0 +1,28 @@
+package timesync
+
+import "time"
+
+type ManualClock struct {
+	*Ticker
+	genesisTime time.Time
+}
+
+func (t *ManualClock) Tick() {
+	t.Notify()
+}
+
+func NewManualClock(genesisTime time.Time) *ManualClock {
+	t := &ManualClock{
+		Ticker:      NewTicker(RealClock{}, LayerConv{genesis: genesisTime}),
```

Comment:
I'm surprised this works... If the layer duration is 0, doesn't it panic when you call `TimeToLayer()` (indirectly, when you call `Notify()`)? Is this tested?

Have you run @antonlerner's `multi_node_sim` (the only place this is used)?

Reply:
Fixed and moved ManualClock to multi nodes.